src/Graphs/partitionedgraphs/partitionedgraph.jl

```diff
@@ -0,0 +1,137 @@
+struct PartitionedGraph{V,PV,G<:AbstractGraph{V},PG<:AbstractGraph{PV}} <:
+       AbstractPartitionedGraph{V,PV}
+  graph::G
+  partitioned_graph::PG
+  partitioned_vertices::Dictionary
+  which_partition::Dictionary
+end
+
+##Constructors.
+function PartitionedGraph(g::AbstractGraph, partitioned_vertices)
+  pvs = keys(partitioned_vertices)
+  pg = NamedGraph(pvs)
+  which_partition = Dictionary()
+  for v in vertices(g)
+    v_pvs = Set(findall(pv -> v ∈ partitioned_vertices[pv], pvs))
+    @assert length(v_pvs) == 1
+    insert!(which_partition, v, first(v_pvs))
+  end
+
+  for e in edges(g)
+    pv_src, pv_dst = which_partition[src(e)], which_partition[dst(e)]
+    pe = NamedEdge(pv_src => pv_dst)
+    if pv_src != pv_dst && !has_edge(pg, pe)
+      add_edge!(pg, pe)
+    end
+  end
+
+  return PartitionedGraph(g, pg, Dictionary(partitioned_vertices), which_partition)
+end
+
+function PartitionedGraph(partitioned_vertices)
+  return PartitionedGraph(NamedGraph(keys(partitioned_vertices)), partitioned_vertices)
+end
+
+function PartitionedGraph(g::AbstractGraph; kwargs...)
+  partitioned_vertices = partition_vertices(g; kwargs...)
+  return PartitionedGraph(g, partitioned_vertices)
+end
+
+#Needed for interface
+partitioned_graph(pg::PartitionedGraph) = getfield(pg, :partitioned_graph)
+unpartitioned_graph(pg::PartitionedGraph) = getfield(pg, :graph)
+partitioned_vertices(pg::PartitionedGraph) = getfield(pg, :partitioned_vertices)
+which_partition(pg::PartitionedGraph) = getfield(pg, :which_partition)
+function vertices(pg::PartitionedGraph, partition_vert::PartitionVertex)
+  return partitioned_vertices(pg)[parent(partition_vert)]
+end
+function vertices(
+  pg::PartitionedGraph, partition_verts::Vector{V}
+) where {V<:PartitionVertex}
```

Comment:
```suggestion
  pg::PartitionedGraph, partition_verts::Vector{<:PartitionVertex}
)
```
I only use `where` if it is really needed, for example if I am using a parameter inside the function or imposing some constraint between types.